src/ApplicationIO.cpp

```diff
@@ -99,6 +103,7 @@ namespace geopm
 
                 platform_io().get_profileio()->connect();
             }
+            m_application_sampler.connect(m_shm_key);
```

Comment:
Maybe this call should just go in the controller next to where application_io->connect() is called.